### PR TITLE
Switch Markdown engine to markdown-it-py

### LINKED ISSUE #16435: Switch Markdown engine to markdown-it-py
Copying from #16414:

The current Markdown engine does not support [fenced code blocks](https://python-markdown.github.io/extensions/fenced_code_blocks/), so it still won’t work after this change. Python-Markdown’s fenced code support is pretty spotty, and if we want to fix that for good IMO we should switch to another Markdown parser. [markdown-it-py](https://github.com/executablebooks/markdown-it-py) (the parser backing [MyST](https://myst-parser.readthedocs.io/en/latest/using/intro.html)) is a popular choice for [CommonMark](https://commonmark.org/) support, which is much closer to [GitHub-Flavored Markdown](https://github.github.com/gfm/) which almost everyone thinks is the standard Markdown (which is unfortunately because GFM is not standard, but that’s how the world works).

### PR DESCRIPTION
**Pull request by torbjornvatn**
As mentioned in #16435 will switching to markdown-it-py give much more advanced Markdown rendering and make it behave almost like to Github Flavored Markdown

closes: #16435
related:  #19644

### CODE REVIEW

**Comment by torbjornvatn** [comment]
@ferruzzi This one passed the Static checks

File: tests/www/test_utils.py
Comment (ashb) [inline tests/www/test_utils.py]: Don't use asseryEqual, keep as a plain assert please.

Comment (torbjornvatn) [reply tests/www/test_utils.py]: Ok, will do

Comment (torbjornvatn) [reply tests/www/test_utils.py]: @ashb fixed now

**Comment by torbjornvatn** [comment]
@ashb I would ideally find a way to remove the `display: block;` property from a `<summary>` inside a `<display>` to get a ‣ in front of the collapsible content just as here on Github. But I couldn't figure out how to change any of the CSS really 🤷🏼 

Example:

<details>
  <summary>Click to expand!</summary>
  
  ## Heading
  1. A numbered
  2. list
     * With some
     * Sub bullets
</details>

**Comment by uranusjr** [comment]
Any updates on this?

**Comment by torbjornvatn** [comment]
Seems like the conflict in `airflow/www/utils.py` is cause by someone else attempting to do something similar, but with a different Markdown lib than me

**Comment by torbjornvatn** [comment]
It was something else actually, fixed the conflict now

**Comment by uranusjr** [comment]
So do we need to do anything special to `<summary>`’s `display: block`?

**Comment by torbjornvatn** [comment]
> So do we need to do anything special to `<summary>`’s `display: block`?

Are you asking me @uranusjr? I don't really know, do you want me to test it somehow?

**Comment by uranusjr** [comment]
You said above

> I would ideally find a way to remove the `display: block;` property from a `<summary>` inside a `<display>` to get a ‣ in front of the collapsible content just as here on Github.

I was just following that up.

**Comment by ashb** [comment]
@bbovenzi CSS questions - can you help out please

**Comment by torbjornvatn** [comment]
> You said above
> 
> > I would ideally find a way to remove the `display: block;` property from a `<summary>` inside a `<display>` to get a ‣ in front of the collapsible content just as here on Github.
> 
> I was just following that up.

🤦🏼‍♂️ yes of course. Sorry for not remembering my own questions. 

A CSS fix for this would be highly appreciated @bbovenzi The working ‣ for expanding the section is kind of important for the UX

**Comment by torbjornvatn** [comment]
A friendly reminder on this one @bbovenzi 

I can do the CSS changes my self, I just needs some pointers on how

**Comment by bbovenzi** [comment]
> A friendly reminder on this one @bbovenzi
> 
> I can do the CSS changes my self, I just needs some pointers on how

My apologies. I kept missing this PR.

We should be able to manipulate the css for markdown simply by adding something like ```.rich_doc summary {
  display: block;
}
``` to `main.css`

File: airflow/www/static/css/main.css
Comment (torbjornvatn) [inline airflow/www/static/css/main.css:469]: This trick works on both Firefox and Chrome. The ‣ is back 🎉 
Screenshot 2022-03-10 at 08 29 36
Screenshot 2022-03-10 at 08 29 25

Comment (torbjornvatn) [reply airflow/www/static/css/main.css:469]: 👀  @bbovenzi

**Review by bbovenzi — APPROVED** [review]
Nice work on the CSS! I'd definitely like to see another approval by someone more familiar with python before merging.

File: airflow/www/utils.py
Comment (uranusjr) [inline airflow/www/utils.py]:
```suggestion
    return Markup(f'<div class="{css_class}" >{md.render(s)}</div>')
```

File: tests/www/test_utils.py
Comment (uranusjr) [inline tests/www/test_utils.py]: Why change this?

Comment (torbjornvatn) [reply tests/www/test_utils.py]: I don't remember, putting it back now

File: setup.py
Comment (uranusjr) [inline setup.py]:
```suggestion
    'linkify-it-py',
    'markdown-it-py',
    'mdit-py-plugins',
```

Sort alphabetically

**Review by uranusjr — APPROVED** [review]

**Comment by torbjornvatn** [comment]
@uranusjr Sorted the deps and merged in main now

**Comment by bbovenzi** [comment]
@torbjornvatn Looks like there are quite a few CI errors that need to be cleaned up.

**Comment by torbjornvatn** [comment]
@bbovenzi Now the test ran ✅ but the building of the 3.7 image failed.
The error in Github actions wasn't very informative though 

Screenshot 2022-04-28 at 23 18 32

**Comment by uranusjr** [comment]
That happens semi-randomly, let me just restart it.

**Comment by potiuk** [comment]
> @uranusjr Now the 3.8 Image one failed smile

RE-run just the failed job

**Comment by potiuk** [comment]
@torbjornvatn -> can you rebase this one on top of latest main in case it fails?

**Comment by torbjornvatn** [comment]
I keep updating from main @potiuk, but the test 3.7 test fails with ` unauthenticated: User cannot be authenticated with the token provided.`

**Comment by torbjornvatn** [comment]
@potiuk now the 3.8 one failed 🙈

**Comment by potiuk** [comment]
I told you it's temporary - one-at-a-time :).

Some people just bring infra failures by just being around. Maybe you are on of those people 🤣

**Comment by uranusjr** [comment]
The docker-compose quick start is failing because it misses the markdown-it-py package. Maybe there’s some dependency list somewhere we need to fix?

Not sure what’s going on with the Helm Chart tests.

**Comment by potiuk** [comment]
Docker compose quick-start tests test starting airflow via docker compose and check if the 

There is nothing "special" about them they are the usual "integration tests" run by `pytest` using breeze environment (mainly to get the right dependencies):

It's rather easy to follow what they do:

Started by `breeze docker-compose tests`
https://github.com/apache/airflow/blob/749e53def43055225a2e5d09596af7821d91b4ac/.github/workflows/ci.yml#L1417

Here are the tests (standard pytest-run python code):

https://github.com/apache/airflow/blob/main/docker_tests/test_docker_compose_quick_start.py

They are also super easy to reproduce and run - you just need to follow what is done in the ci.

There is no magic there. If they do not start, it means that something does not work and you cannot inspect airflow after you use production image to ru docker-compose with airflow.

**Comment by potiuk** [comment]
Those tests are converted to run standard "breeze" commands and you can always easyly locally reproduce it using breeze.

**Comment by potiuk** [comment]
I just recommend to try to reproduce it following it step-by-step.

**Comment by torbjornvatn** [comment]
@potiuk and @uranusjr I'm having a really hard time getting all these checks to be ✅ at the same time 

The last one was something with the python cache or something, and I'm not allowed to rerun individual workflows either as far as I can see.

Screenshot 2022-06-21 at 09 05 12

**Comment by potiuk** [comment]
Intermittent problem. I just re-run failed jobs only. 

BTW. There is the "re-run failed jobs" button at the top (not sure if you can do it as non-committer though).

**Comment by torbjornvatn** [comment]
🎉 @potiuk 
It worked

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Woooohooo :tada: :tada: :tada: :tada: :tada: :tada:

File: airflow/www/static/css/main.css
Comment (tronlightracer) [inline airflow/www/static/css/main.css:473]: I'm no css expert but the values should be in 'em' not 'px' so the display can be dynamically based off the screen size of the viewer.

Comment (potiuk) [inline airflow/www/static/css/main.css:473]: Maybe you should provide and test PR with a fix @tronlightracer ? This is the best way to become one of > 2100 contributors to Airflow (mostly people like you).

**Pull request merged**
